server/service/http/forward.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package http
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/CeresDB/ceresmeta/pkg/log"
+	"github.com/CeresDB/ceresmeta/server/member"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type ForwardClient struct {
+	member *member.Member
+	client *http.Client
+	port   int
+}
+
+func NewForwardClient(member *member.Member, port int) *ForwardClient {
+	return &ForwardClient{
+		member: member,
+		client: getForwardedHTTPClient(),
+		port:   port,
+	}
+}
+
+func getForwardedHTTPClient() *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).DialContext,
+			TLSHandshakeTimeout: 10 * time.Second,
+		},
+	}
+}
+
+func (s *ForwardClient) getForwardedAddr(ctx context.Context) (string, bool, error) {
```

Comment:
There is some duplication of code consistent with the forwarding logic in grpc. Define some commonly used methods to avoid duplication of code.

Reply:
The duplicated code is `member.GetLeader` and valid logic of it's result, I think it is no need to add common method for it.